### ISSUE TITLE
Enable treesitter capture groups code chunks in `.qmd` files

### ISSUE THREAD

**Issue by nsfinkelstein** [quarto-dev/quarto-nvim#87]
Tree sitter provides shortcuts to select meaningful chunks of text. For example, the `@function.outer` capture group selects the function (including definition) in which the cursor is contained. These capture groups can be mapped to keys (traditionally `af` for `@function.outer`).

The capture groups do not seem to work in `.qmd` files.

To reproduce, using the following standard treesitter neovim setup:

```
require('nvim-treesitter.configs').setup {
  ensure_installed = {
    'python',
  },
  highlight = {
    enable = true, -- false will disable the whole extension
  },
  incremental_selection = {
    enable = true,
    keymaps = {
      init_selection = 'gnn',
      node_incremental = 'grn',
      scope_incremental = 'grc',
      node_decremental = 'grm',
    },
  },
  indent = {
    enable = true,
  },
  textobjects = {
    select = {
      enable = true,
      lookahead = true, -- Automatically jump forward to textobj, similar to targets.vim
      keymaps = {
        -- You can use the capture groups defined in textobjects.scm
        ['af'] = '@function.outer',
        ['if'] = '@function.inner',
        ['ac'] = '@class.outer',
        ['ic'] = '@class.inner',
      },
    },
    move = {
      enable = true,
      set_jumps = true, -- whether to set jumps in the jumplist
      goto_next_start = {
        [']m'] = '@function.outer',
        [']]'] = '@class.outer',
      },
      goto_next_end = {
        [']M'] = '@function.outer',
        [']['] = '@class.outer',
      },
      goto_previous_start = {
        ['[m'] = '@function.outer',
        ['[['] = '@class.outer',
      },
      goto_previous_end = {
        ['[M'] = '@function.outer',
        ['[]'] = '@class.outer',
      },
    },
  },
}
```

Create a python file with the following:

```
def test_func():
    return 123
```

Pressing `yaf` should copy the function, as can be confirmed by pressing `p` to paste it.

If the same function is included in `python` code block in a `.qmd` file, pressing `yaf` does not copy it.

Thank you for any help or advice.

**Comment by jmbuhr** [comment]
I can reproduce this and found the reason for it: https://github.com/nvim-treesitter/nvim-treesitter-textobjects/issues/131
tldr:
> They only work when the parent language has a textobject of that kind. The keybindings will only be set for the parent language because language injection is dynamic which mean any language could be injected at any moment. [...]
> 
> You could set the keybindings manually. Eg `TSTexobjectSelect @paramater.outer` should work.
https://github.com/nvim-treesitter/nvim-treesitter-textobjects/issues/131#issuecomment-987122673

**Comment by jmbuhr** [comment]
Another solution is to do what I just did in the dev branch of my config, see commit above (https://github.com/jmbuhr/quarto-nvim-kickstarter/commit/0678967327306544168de040611d219d4723e8db)

**Comment by nsfinkelstein** [comment]
Thanks for the pointer! Closing this issue because it seems like a treesitter issue.

**Comment by jmbuhr** [comment]
I will leave it open until I have at least pushed the workaround to main of the kickstarter configuration to remind myself :)
